manifests/init.pp

```diff
@@ -260,7 +269,29 @@
     validate_absolute_path($keys_real)
   }
 
-  validate_string($restrict_options)
+  if is_array($restrict_options) == true {
+    $restrict_options_real = $restrict_options
+  }
+  # needed for backward compatibility
+  elsif is_string($restrict_options) == true {
+    $restrict_options_real = $restrict_options ? {
+      'USE_DEFAULTS' => $default_restrict_options,
+      default        => [ "-4 $restrict_options", "-6 $restrict_options", ]
+    }
+  }
+  else {
+    fail("restrict_options must be an array (prefered) or a string (will be deprecated).")
```

Comment:
Totally agree. If we bump the major version, we can introduce other breaking changes. Is there anything else we should change? I was thinking of merging this and your solaris code and making that the last of v3 and then we could move to v4, but I do not want to do that unless there are really compelling reasons to break compatibility.